pkg/cdc/generate_config.go

```diff
@@ -0,0 +1,124 @@
+		}
+	}
+	// Marshall everything back to json and mutate the input.
+	marshalled, err := json.Marshal(c)
+	if err != nil {
+		return nil, err
+	}
+	return marshalled, nil
+}
+
+// updateAdditionalJVMOpts adds CDC related entries to additional-jvm-opts. Docs here https://docs.datastax.com/en/cdc-for-cassandra/cdc-apache-cassandra/$%7Bversion%7D/index.html
+func updateAdditionalJVMOpts(optsSlice []string, CDCConfig *cassdcapi.CDCConfiguration) ([]string, error) {
+	out := removeEntryFromSlice(optsSlice, "pulsarServiceUrl")
+	//Next, create an additional options entry that instantiates the settings we want.
+	if CDCConfig == nil {
+		return optsSlice, nil
+	} else {
+		reflectedCDCConfig := reflect.ValueOf(*CDCConfig)
+		t := reflectedCDCConfig.Type()
+		optsSlice := []string{}
+		for i := 0; i < reflectedCDCConfig.NumField(); i++ {
+			// This logic depends on the json tags from the CR mapping to the CDC agent's parameter names.
+			fieldName := t.Field(i).Name
+			t := reflect.TypeOf(*CDCConfig)
+			reflectedField, ok := t.FieldByName(fieldName)
+			if !ok {
+				return nil, errors.New(fmt.Sprint("could not get CDC field", fieldName))
+			}
+			nameTag := strings.Split(string(reflectedField.Tag.Get("json")), ",")[0]
+			reflectedValue := interface{}(nil)
+			// We need to get value types back from pointer types here and handle nil pointers.
+			switch reflectedField.Type.Kind() {
+			case reflect.Ptr:
+				if !reflectedCDCConfig.Field(i).IsNil() { // We only want to append the value if it is non-nil
+					reflectedValue = reflectedCDCConfig.Field(i).Elem().Interface()
+					optsSlice = append(optsSlice, nameTag+"="+fmt.Sprintf("%s", reflectedValue))
+				}
+			case reflect.Array:
+				return nil, fmt.Errorf("invalid type in CDC struct, cannot reflect field %s", fieldName)
+			case reflect.Chan:
+				return nil, fmt.Errorf("invalid type in CDC struct, cannot reflect %s", fieldName)
+			case reflect.Func:
+				return nil, fmt.Errorf("invalid type in CDC struct, cannot reflect %s", fieldName)
+			case reflect.Map:
+				return nil, fmt.Errorf("invalid type in CDC struct, cannot reflect %s", fieldName)
+			case reflect.Interface:
+				return nil, fmt.Errorf("invalid type in CDC struct, cannot reflect %s", fieldName)
+			case reflect.Slice:
+				return nil, fmt.Errorf("invalid type in CDC struct, cannot reflect %s", fieldName)
+			default: // no need to call .Elem() when we have a value type.
+				reflectedValue = reflectedCDCConfig.Field(i).Interface()
+				optsSlice = append(optsSlice, nameTag+"="+fmt.Sprintf("%s", reflectedValue))
+			}
+		}
+		CDCOpt := fmt.Sprintf("-javaagent:%s=%s", "/opt/cdc_agent/cdc-agent.jar", strings.Join(optsSlice, ","))
+		out = append(out, // We need to add these two elements to the slice first, because the management agent must start before the CDC agent.
+			"-javaagent:/opt/metrics-collector/lib/datastax-mcac-agent.jar",
+			"-javaagent:/opt/management-api/datastax-mgmtapi-agent-0.1.0-SNAPSHOT.jar",
```

Comment:
If https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/94 is ever fixed (and it was intended), then this will blow up. If we ever remove MCAC, then that would cause this to blow up also.

Reply:
Shouldn't this kind of logic live in the management-api .sh scripts to ensure things work? Instead of modifying it here, modify it there based on some ENV variable? ``ENABLE_CDC=true`` ?